templates/skeleton/app/root.tsx

```diff
@@ -6,15 +6,25 @@ import {
   Outlet,
   Scripts,
   useRouteError,
-  useLoaderData,
+  useRouteLoaderData,
   ScrollRestoration,
   isRouteErrorResponse,
   type ShouldRevalidateFunction,
 } from '@remix-run/react';
+import type {
+  CartApiQueryFragment,
+  FooterQuery,
+  HeaderQuery,
+} from 'storefrontapi.generated';
 import favicon from './assets/favicon.svg';
 import resetStyles from './styles/reset.css?url';
 import appStyles from './styles/app.css?url';
-import {Layout} from '~/components/Layout';
+import {Layout as PageLayout} from '~/components/Layout';
+
+export type RootLoader = typeof loader;
+export type RootLoaderData = NonNullable<
+  ReturnType<typeof useRouteLoaderData<RootLoader>>
+>;
```

Comment:
Nice! Maybe we can even remove `RootLoaderData` if we rely more on query/fragment types, which are a bit safer I think?